go-controller/pkg/ovn/base_network_controller_policy.go

```diff
@@ -520,7 +520,7 @@ func (bnc *BaseNetworkController) getNewLocalPolicyPorts(np *networkPolicy,
 
 		// Skip pods that will never be present in logicalPortCache,
 		// e.g. hostNetwork pods, overlay node pods, or completed pods
-		if !bnc.podExpectedInLogicalCache(pod) {
+		if !bnc.podExpectedInLogicalCache(pod) && bnc.doesNetworkRequireIPAM() {
```

Comment:
Let's update `podExpectedInLogicalCache` to return false only for pods that won't be found in the `logicalPortCache`

Reply:
Done.

The better term fix would be to break the assumption a logical switch without a subnet represents an host network.